### PR TITLE
oldisim integration code

### PR DESCRIPTION
**Pull request by huinghuang**
The following changes are made:

add oldisim to README.md.
add oldisim_benchmark.py to /benchmarks
add oldisim_dependencies.py to /packages

### CODE REVIEW

File: perfkitbenchmarker/benchmarks/oldisim_benchmark.py [hunk 1]
```diff
@@ -0,0 +1,253 @@
...
+import time
+
+from perfkitbenchmarker import flags
+from perfkitbenchmarker import sample
+from perfkitbenchmarker import vm_util
+
+
+FLAGS = flags.FLAGS
+
+flags.DEFINE_integer('num_leaves', 4, 'number of leaf nodes',
```
Comment (cmccoy) [inline perfkitbenchmarker/benchmarks/oldisim_benchmark.py]: Would you mind prefixing these flags with the benchmark name (e.g. `oldisim_num_leaves`), so it's clear what benchmark they affect?

File: perfkitbenchmarker/benchmarks/oldisim_benchmark.py [hunk 2]
```diff
@@ -0,0 +1,253 @@
...
+flags.DEFINE_list('fanout', [],
+                  'a list of fanouts to be tested. '
+                  'a root can connect to a subset of leaf nodes (fanout). '
+                  'the value of fanout has to be smaller than num_leaves.')
+flags.DEFINE_enum('latency_metric', 'avg',
+                  ['avg', '50p', '90p', '95p', '99p', '99.9p'],
+                  'Allowable metrics for end-to-end latency')
+flags.DEFINE_float('latency_target', '30', 'latency target in ms')
+
+OLDISIM_GIT = 'https://github.com/GoogleCloudPlatform/oldisim.git'
```
Comment (cmccoy) [inline perfkitbenchmarker/benchmarks/oldisim_benchmark.py]: Please also specify a git revision here and check it out below.

File: perfkitbenchmarker/benchmarks/oldisim_benchmark.py [hunk 3]
```diff
@@ -0,0 +1,253 @@
...
+
+  # Launch job on the leaf nodes.
+  leaf_server_bin = os.path.join(OLDISIM_DIR, BINARY_BASE, 'LeafNode')
+  for vm in leaf_vms:
+    leaf_cmd = '%s --threads=%s' % (leaf_server_bin, vm.num_cpus)
+    vm.RemoteCommand('%s &> /dev/null &' % leaf_cmd)
+
+
+def SetupRoot(root_vm, leaf_vms):
+  """Connect a root node to a list of leave nodes.
```
Comment (cmccoy) [inline perfkitbenchmarker/benchmarks/oldisim_benchmark.py]: grammar nit: leave -> leaf.

File: perfkitbenchmarker/benchmarks/oldisim_benchmark.py [hunk 4]
```diff
@@ -0,0 +1,253 @@
...
+  """Run Loadtest for a given topology.
+
+  Args:
+    benchmark_spec: The benchmark specification. Contains all data that is
+        required to run the benchmark.
+    fanout: Request is first processed by a root node, which then
+        fans out to a subset of leaf nodes.
+
+  Returns:
+    A list of samples in the form of 3 or 4 tuples. The tuples contain
```
Comment (cmccoy) [inline perfkitbenchmarker/benchmarks/oldisim_benchmark.py]: This doesn't look right.

File: perfkitbenchmarker/benchmarks/oldisim_benchmark.py [hunk 5]
```diff
@@ -0,0 +1,253 @@
...
+    A tuple of (peak_qps, peak_lat, target_qps, target_lat).
+  """
+
+  re_peak = re.compile(r'peak qps = (?P<qps>\S+), latency = (?P<lat>\S+)')
+  re_target = re.compile(r'measured_qps = (?P<qps>\S+), latency = (?P<lat>\S+)')
+
+  for line in oldisim_output.splitlines():
+    match = re.search(re_peak, line)
+    if match:
+      peak_qps = match.group('qps')
```
Comment (cmccoy) [inline perfkitbenchmarker/benchmarks/oldisim_benchmark.py]: Recommend converting to floats here.

**Comment by cmccoy** [comment]
LGTM.

**Comment by cmccoy** [comment]
Thanks, @huinghuang! Merge whenever you're ready.

**Comment by voellm** [comment]
Awesome... this will make this coming Mondays release train.

**Pull request merged**